terkin/sensor/core.py

```diff
@@ -386,8 +386,9 @@ def power_on(self):
             self.just_started = False
             return
 
-        # TODO: Add implementation for Vanilla MicroPython.
-        self.adapter.init(mode=I2C.MASTER, baudrate=self.frequency)
+        # uPy doesn't have deinit so it doesn't need init
+        if platform_info.vendor == platform_info.MICROPYTHON.Pycom:
+            self.adapter.init(mode=I2C.MASTER, baudrate=self.frequency)
```

Comment:
Isn't there something how this is done on Genuine MicroPython on Vanilla ESP32?

Reply:
uPys I2C has no/doesn't need init/deinit so AFAIK nothing can be done.